python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -536,3 +563,150 @@ def to_arrow(self):
+        the same.
+
+        This implements `.contains_properly`, which shares a large
+        space of correct cases with `GeoPandas.contains` but they do not
+        produce identical results. In the future we will use intersection
+        testing to match .contains behavior.
+
+        Parameters
+        ----------
+        test_points_x
+            x-coordinate of test points
+        test_points_y
+            y-coordinate of test points
+        poly_offsets
+            beginning index of the first ring in each polygon
+        poly_ring_offsets
+            beginning index of the first point in each ring
+        poly_points_x
+            x closed-coordinate of polygon points
+        poly_points_y
+            y closed-coordinate of polygon points
+
+        Examples
+        --------
+
+        Test if a polygon is inside another polygon:
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0.5, 0.5)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        dtype: bool
+
+
+        Test whether 3 points fall within either of two polygons
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0, 0)],
+            [Point(0, 0)],
+            [Point(0, 0)],
+            [Point(-1, 1)],
+            [Point(-1, 1)],
+            [Point(-1, 1)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        1    False
+        2    False
+        3     True
+        4     True
+        5     True
+        dtype: bool
+
+        Note
+        ----
+        input Series x and y will not be index aligned, but computed as
+        sequential arrays.
+
+        Note
+        ----
+        poly_ring_offsets must contain only the rings that make up the polygons
+        indexed by poly_offsets. If there are rings in poly_ring_offsets that
+        are not part of the polygons in poly_offsets, results are likely to be
+        incorrect and behavior is undefined.
+
+        Returns
+        -------
+        result : cudf.DataFrame
+            A DataFrame of boolean values indicating whether each point falls
+            within each polygon.
```

Comment:
It is unclear whether broadcasting is occuring. Is this pairwise? Does the number of test points have to equal the number of polygons as in the two examples?  "each point ... each polygon" is ambiguous, as it could mean pairwise (each pair) or all pairs (all combinations of pairs).

if this is pairwise, then please clarify above and suggest using the word "corresponding" here: 

```suggestion
            A DataFrame of boolean values indicating whether each point falls
            within the corresponding polygon in the input.
```

Reply:
It is pairwise at the user level, yes. Thanks!